### PR TITLE
Add filtering for consumer groups and make results case insensitive

### PR DESCRIPTION
**Pull request by suyash-naithani**

### CODE REVIEW

**Comment by wtrocki** [comment]
Amazing change! @suyash-naithani thank you so much! I will rebase your PR right now.
Have you verified this change with kafka-ui?

File: packages/kafka-instance-sdk/src/generated/api/default-api.ts [hunk 1]
```diff
@@ -213,7 +213,7 @@ export const DefaultApiAxiosParamCreator = function (configuration?: Configurati
          * @param {*} [options] Override http request option.
          * @throws {RequiredError}
          */
-        getConsumerGroups: async (offset?: number, limit?: number, size?: number, page?: number, topic?: string, groupIdFilter?: string, order?: 'asc' | 'desc', orderKey?: 'name', options: any = {}): Promise<RequestArgs> => {
+         getConsumerGroups: async (offset?: number, limit?: number, filter?:string ,size?: number, page?: number, topic?: string, groupIdFilter?: string, order?: 'asc' | 'desc', orderKey?: 'name', options: any = {}): Promise<RequestArgs> => {
```
Comment (wtrocki) [inline packages/kafka-instance-sdk/src/generated/api/default-api.ts:216]: Question - why we have done this change. This is generated code from openapi. If we are missing filter this change should be actually made in the backend openapi file.

Comment (wtrocki) [reply packages/kafka-instance-sdk/src/generated/api/default-api.ts:216]: This argument is not needed. There is `groupIdFilter`. I do not see filter present in OpenAPI or in code. 

CC @MikeEdgar

Comment (wtrocki) [reply packages/kafka-instance-sdk/src/generated/api/default-api.ts:216]: Relevant PR:  https://github.com/bf2fc6cc711aee1a0c2a/kafka-admin-api/pull/48

Comment (suyash-naithani) [reply packages/kafka-instance-sdk/src/generated/api/default-api.ts]: Thanks for pointing that out @wtrocki. I have made some changes to my PR. Let me know what you think

**Review by wtrocki — APPROVED** [review]

**Pull request merged**